File: API/CustomerManagementSystemApi/CustomerManagementSystem.Domain/Models/AuditLogEntry.cs
namespace CustomerManagementSystem.Domain.Models;

public sealed record AuditLogEntry
{
    public required int CustomerAuditLogId { get; init; }

    public required Guid CustomerId { get; init; }

    public required string PerformedBy { get; init; }

    public required AuditAction ActionType { get; init; }

    public string? Details { get; init; }

    public required DateTime OccurredAt { get; init; }
}

public sealed record GlobalAuditLogEntry
{
    public required int CustomerAuditLogId { get; init; }

    public required Guid CustomerId { get; init; }

    public string? CustomerFirstName { get; init; }

    public string? CustomerLastName { get; init; }

    public required string PerformedBy { get; init; }

    public required AuditAction ActionType { get; init; }

    public string? Details { get; init; }

    public required DateTime OccurredAt { get; init; }
}

File: UI/src/app/interfaces/audit-log-entry.ts
import { IsoDateTime } from './iso-date';

export type AuditAction =
  | 'Created'
  | 'Edited'
  | 'Deactivated'
  | 'Reactivated'
  | 'Deleted'
  | 'Purchased';

export interface AuditLogEntry {
  customerAuditLogId: number;
  customerId: string;
  performedBy: string;
  actionType: AuditAction;
  details: string | null;
  occurredAt: IsoDateTime;
}
